tensorrt_llm/visual_gen/visual_gen.py

```
@@ -527,8 +474,70 @@ def __init__(
+        """Resolve pipeline class from checkpoint for extra_param_specs discovery."""
+        try:
+            from tensorrt_llm._torch.visual_gen.pipeline_registry import AutoPipeline
+
+            checkpoint_dir = self.args.checkpoint_path
+            if checkpoint_dir:
+                pipeline_type = AutoPipeline._detect_from_checkpoint(checkpoint_dir)
+                from tensorrt_llm._torch.visual_gen.pipeline_registry import PIPELINE_REGISTRY
+
+                return PIPELINE_REGISTRY.get(pipeline_type)
+        except Exception:
+            pass
```

Ruff (0.15.9)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)